packages/babel-core/src/transformation/normalize-file.js

```diff
@@ -28,9 +28,16 @@ export default function normalizeFile(
   let shebang = null;
   let inputMap = null;
   if (options.inputSourceMap !== false) {
-    inputMap = convertSourceMap.fromSource(code);
+    const inputMapSource = convertSourceMap.fromSource(code);
+    inputMap =
+      inputMapSource !== null
+        ? inputMapSource
+        : convertSourceMap.fromMapFileSource(code);
```

Comment:
Yeah that definitely looks cleaner. I am just thinking is there a reason we would wrap the `fromMapFileSource` in a try/catch and not the `fromSource` as well?

Reply:
Sure, wrapping both sounds good. I'm less concerned about it because it's existing code and because it will only throw if there is a `data:` URL with a badly-made sourcemap, whereas it's much more likely that there will be a file that links to a file that doesn't exists.